packages/office-ui-fabric-react/src/components/FloatingPicker/PeoplePicker/examples/FloatingPeoplePicker.Basic.Example.tsx

```diff
@@ -10,128 +9,96 @@ import {
+
+const getTextFromItem = (persona: IPersonaProps): string => {
+  return persona.text || '';
+};
+const listContainsPersona = (persona: IPersonaProps, personas?: IPersonaProps[]): boolean => {
+  return !!personas && personas.some((item: IPersonaProps) => item.text === persona.text);
+};
+const validateInput = (input: string): boolean => {
+  return input.indexOf('@') !== -1;
+};
+const startsWith = (text: string, filterText: string): boolean => {
+  return text.toLowerCase().indexOf(filterText.toLowerCase()) === 0;
+};
 
-  private _onFocus = (): void => {
-    if (this._picker.current) {
-      this._picker.current.showPicker();
-    }
-  };
+export const FloatingPeoplePickerTypesExample: React.FunctionComponent = () => {
+  const inputElementRef = React.useRef<HTMLInputElement>(null);
+  const [peopleList, setPeopleList] = React.useState(people);
+  const [searchValue, setSearchValue] = React.useState('');
+  const picker = React.useRef<IBaseFloatingPicker>(null);
 
-  private _setInputElementRef = (ref: HTMLDivElement | null) => {
-    const inputElements = ref && ref.getElementsByClassName('ms-SearchBox-field');
-    if (inputElements && inputElements.length > 0) {
-      this._inputElement = inputElements[0] as HTMLInputElement;
+  const onFocus = (): void => {
+    if (picker.current) {
+      picker.current.showPicker();
     }
   };
 
-  private _onSearchChange = (ev: React.ChangeEvent<HTMLInputElement>, newValue: string): void => {
-    if (newValue !== this.state.searchValue && this._picker.current) {
-      this.setState({ searchValue: newValue });
-      this._picker.current.onQueryStringChanged(newValue);
+  const onSearchChange = (ev: React.ChangeEvent<HTMLInputElement>, newValue: string): void => {
+    if (newValue !== searchValue && picker.current) {
+      setSearchValue(newValue);
+      picker.current.onQueryStringChanged(newValue);
     }
   };
 
-  private _onPickerChange = (selectedSuggestion: IPersonaProps): void => {
-    this.setState({ searchValue: selectedSuggestion.text || '' });
-    if (this._picker.current) {
-      this._picker.current.hidePicker();
+  const onPickerChange = (selectedSuggestion: IPersonaProps): void => {
+    setSearchValue(selectedSuggestion.text || '');
+    if (picker.current) {
+      picker.current.hidePicker();
     }
   };
 
-  private _onRemoveSuggestion = (item: IPersonaProps): void => {
-    const { peopleList } = this.state;
+  const onRemoveSuggestion = (item: any): void => {
     const itemIndex = peopleList.indexOf(item);
-
     if (itemIndex >= 0) {
-      this.setState({
-        peopleList: peopleList.slice(0, itemIndex).concat(peopleList.slice(itemIndex + 1)),
-      });
+      setPeopleList(peopleList.slice(0, itemIndex).concat(peopleList.slice(itemIndex + 1)));
     }
   };
 
-  private _onFilterChanged = (filterText: string, currentPersonas?: IPersonaProps[]): IPersonaProps[] => {
+  const suggestionProps: IBaseFloatingPickerSuggestionProps = {
```

Comment:
This should probably go inside a `useConst` to avoid creating a new object every time (it can be a const because the `picker` ref object itself won't change, even if the thing inside does)